jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -261,6 +248,28 @@ public ImagesAndRegistryClient call()
     }
   }
 
+  private Optional<ImagesAndRegistryClient> getImagesAndRegistryClient(
+      EventHandlers eventHandlers,
+      String mirror,
+      ProgressEventDispatcher progressDispatcher2,
+      RegistryClient registryClient)
+      throws IOException, RegistryException, LayerCountMismatchException,
+          BadContainerConfigurationFormatException {
+    try {
+      List<Image> images = pullBaseImages(registryClient, progressDispatcher2.newChildProducer());
+      eventHandlers.dispatch(LogEvent.info("pulled manifest from mirror " + mirror));
+      return Optional.of(new ImagesAndRegistryClient(images, registryClient));
+
+    } catch (RegistryUnauthorizedException ex) {
+      // in case if a mirror requires bearer auth
+      eventHandlers.dispatch(LogEvent.debug("mirror " + mirror + " requires auth"));
+      registryClient.doPullBearerAuth();
+      List<Image> images = pullBaseImages(registryClient, progressDispatcher2.newChildProducer());
+      eventHandlers.dispatch(LogEvent.info("pulled manifest from mirror " + mirror));
+      return Optional.of(new ImagesAndRegistryClient(images, registryClient));
+    }
+  }
```

Comment:
```suggestion
  private List<Image> pullPublicImages(
      EventHandlers eventHandlers,
      RegistryClient registryClient,
      ProgressEventDispatcher progressDispatcher)
      throws IOException, RegistryException, LayerCountMismatchException,
          BadContainerConfigurationFormatException {
    try {
      // First, try with no credentials. This works with public GCR images.
      return pullBaseImages(registryClient, progressDispatcher.newChildProducer());

    } catch (RegistryUnauthorizedException ex) {
      // in case if a registry requires bearer auth
      registryClient.doPullBearerAuth();
      return pullBaseImages(registryClient, progressDispatcher.newChildProducer());
    }
  }
```